libstb/secvar/backend/backend-common.c

```diff
@@ -0,0 +1,142 @@
+// SPDX-License-Identifier: Apache-2.0
+/* Copyright 2019 IBM Corp. */
+#include <opal.h>
+#include <string.h>
+#include <time.h>
+#include <unistd.h>
+#include <stdint.h>
+#include <device.h>
+#include <ccan/endian/endian.h>
+#include <mbedtls/error.h>
+#include "backend-common.h"
+#include "../secvar.h"
+
+bool setup_mode;
+
+int add_to_variable_bank(struct secvar *secvar, const char *data, uint64_t dsize)
+{
+	struct secvar_node *node;
+
+	node = find_secvar(secvar->key, secvar->key_len, &variable_bank);
```

Comment:
Is it possible to pass around a `struct secvar_node` more often rather than just the `data`? Would prevent an O(n) lookup each time this function is called.

Reply:
This I will look at the end after fixing all the other changes.. in one go through..

Reply:
I think I can just pass the struct secvar_node rather than struct secvar. This will save one call each time this function is called.

Reply:
And as I tried to do that.. I realized.. I can't.. as secvar_node can be from update bank.. which tells what key to be used.. But what needs to be updated is in variable bank